web/src/Root.jsx

```diff
@@ -176,21 +176,7 @@ class Root extends Component {
     }
   }
 
-  tick = () => {
-    if (!this.state.connectionTerminated) {
-      clearInterval(this.timer);
-      return;
-    }
-    if (this.state.seconds > 0) {
-      this.setState({ seconds: this.state.seconds - 1 });
-    } else {
-      this.setState({ seconds: 9 });
-      clearInterval(this.timer);
-    }
-  }
-
   ping = async () => {
```

Comment:
why does this still exist in Root, shouldn't ConnectTerminated component take care of everything?

Reply:
@salahalsaleh1993  No, that component is only rendered when there is no connection and manages checking from there. Once a connection is reestablished it falls back to root to ping every 10 seconds to check if there is still a connection.

Reply:
Aha gotcha, would it make sense to move the modal to be inside `ConnectionTerminated` and let that component handle everything?

Reply:
Might be? I was trying to avoid having that component rendered all the time. Also would need to spend more time refactoring to make sure polling stops when connection is terminated so the checks can happen at the right interval and then start polling every 10 seconds again after it's established. I had already spent more time that I was planning with a different approach that ended up not really working at all so I didn't want to add even more time refactoring so this new component handled it all. Could probably go back with this new approach and make it all in one component with out too much issue now. Might look into that a little later today.